submissions/Harnytskyi/friends-app/js/script.js

```diff
@@ -0,0 +1,163 @@
+            <p class="name">${item.name}</p>
+            <p class="age">Age: ${item.age}</p>
+            <a class="phone" href="tel:${item.phone}">${item.phone}</a>
+            <div class="email">
+                <a href="mailto:${item.email}">
+                    <button class="button-message">SEND MESSAGE</button>
+                </a>
+            </div>
+        </div>`
+    }).join("");
+}
+
+function changeFilterStatus(checkbox) {
+    if (checkbox.checked) {
+        gender[checkbox.value] = true;
+    } else {
+        if (checkbox.value == "male") {
+            if (gender.female == false) {
+                gender.female = true;
+                CHECKBOX_WOMAN.click();
+            }
+        }
+        else if (checkbox.value == "female") {
+            if (gender.male == false) {
+                gender.male = true;
+                CHECKBOX_MAN.click();
+            }
+        }
+        gender[checkbox.value] = false;
+    }
+    filterByGender();
+    displayFriends();
+}
+function searchEvent() {
+    filterByGender();
+    displayFriends();
+}
+
+function search() {
+    let searchRequest = SEARCH_INPUT.value.toUpperCase();
+    searchedFriends = [];
+    selectedFriends.forEach(item => {
+        if (item.name.toUpperCase().includes(searchRequest)) {
+            searchedFriends.push(item)
+        }
+        selectedFriends = searchedFriends;
+    })
+}
+
+function sortFriends(value) {
+    let criterion;
+    let order;
+    switch (value) {
+        case 'alphabet-ascend':
+            criterion = 'name';
+            break;
+        case 'alphabet-descend':
+            criterion = 'name';
+            order = 'descending';
+            break;
+        case 'age-ascend':
+            criterion = 'age';
+            break;
+        case 'age-descend':
+            criterion = 'age';
+            order = 'descending';
+            break;
+    }
+    friends.sort((a, b) => {
+        if (order == "descending") {
+            let c = a;
+            a = b;
+            b = c;
+        }
+        if (a[criterion] > b[criterion]) return 1;
+        if (a[criterion] == b[criterion]) return 0;
+        if (a[criterion] < b[criterion]) return -1;
+    })
+    filterByGender();
+    displayFriends();
+}
+
+FILTER_GENDER.addEventListener('change', (event) => {
+    let target = event.target;
+    changeFilterStatus(target);
+})
+FILTER_SORT.addEventListener('change', (event) => {
+    let target = event.target;
+    sortFriends(target.value);
+})
+
+function fetchUsers() {
+    let targetUrl = "https://randomuser.me/api/?results=30&nat=au,ca,ch,de,dk,es,fr,gb,ie,no,nl,nz,us"
+
+    fetch(targetUrl, { metod: "get" })
+        .then((response) => {
+            let json = response.json();
+            if (response.ok) {
+                return json;
+            }
+            else {
+                return error;
+            }
+        }).then((json) => {
+            saveUsers(json.results);
+        })
```

Comment:
Single responsibility - your functions should do only one job. As an example function in which you fetch users should only fetch them and not render, transform or process them in other ways. All these things should be done in another place, outside your function. The same applied to the sort function, which usually does all types of sorting 😉